WIP: DFROBOT SEN0623 (C1001)

ennetech opened the pull request:
Work in progress for C1001, base protocol implemented, and stubs for almost all has been added


Product page: https://wiki.dfrobot.com/SKU_SEN0623_C1001_mmWave_Human_Detection_Sensor
Integration reference: https://github.com/DFRobot/DFRobot_HumanDetection

- [X] Wrap protocol
- [ ] Integrate setup configuration sync
- [ ] Implement all options for fall mode
- [ ] Implement all options for sleep mode
- [X] Mode switching
- [ ] Data dump
- [X] Add data-pooling

Bugs:
- [ ] Sometimes if no presence is detected and the device stops sending data in the UART it's possible that a crash occurs

```
esphome:
  name: esp32-c1001
  friendly_name: ESP32-C1001

esp32:
  board: esp32dev
  framework:
    type: esp-idf

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: !secret encryption_key

ota:
  - platform: esphome
    password: !secret ota_password

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Fallback Hotspot"
    password: !secret fallback_password

captive_portal:

web_server:
  port: 80

# custom components
external_components:
  - source:
      type: git
      url: https://github.com/ennetech/esphome
      ref: component/dfrobot_sen0623
    components: [ dfrobot_sen0623 ]

# device config
uart:
  tx_pin: 16
  rx_pin: 17
  baud_rate: 115200

# base component
dfrobot_sen0623:
  update_interval: 5s

sensor:
  - platform: dfrobot_sen0623
    heart_rate:
      name: "Heart rate"
    breath_rate:
      name: "Breath rate"
    human_distance:
      name: "Human distance"
    human_move_range:
      name: "Human move range"

text_sensor:
  - platform: dfrobot_sen0623
    status:
      name: "Operating mode"
    movement:
      name: "Movement type"
      
binary_sensor:
  - platform: dfrobot_sen0623
    presence:
      name: "Presence"

switch:
  - platform: dfrobot_sen0623
    type: request_rate
    name: "Request rate"
  - platform: dfrobot_sen0623
    type: hp_led
    name: "HP Led Enabled"
button:
  - platform: dfrobot_sen0623
    type: mode_fall
    name: "Fall mode"
  - platform: dfrobot_sen0623
    type: mode_sleep
    name: "Sleep mode"
  - platform: dfrobot_sen0623
    type: reset
    name: "Reset"
```

ennetech committed:
initial stub

ennetech committed:
added other entities

ennetech committed:
added text sensor

ennetech committed:
rework command system

ennetech committed:
stub for pooling

ennetech committed:
merge all id into same structure

ennetech committed:
move to process & wait

ennetech committed:
cleanup

ennetech committed:
breath_rate_sensor

ennetech committed:
rename motion to presnce

ennetech committed:
fixes in event loop

ennetech committed:
maybe solve crash when sensor stops sending data (after presence change)

ennetech committed:
add movement type text sensor

ennetech committed:
add human distance

ennetech committed:
added human move range

TroySK commented:
Is this ready? How do I use this with my existing ESPHome installation?

ennetech commented:
> Is this ready? How do I use this with my existing ESPHome installation?

Just copy the configuration in the first post, at the moment it exposes the readings but it still need some work

TroySK commented:
> > Is this ready? How do I use this with my existing ESPHome installation?
> 
> Just copy the configuration in the first post, at the moment it exposes the readings but it still need some work

That is awesome! I had overlooked the `external_components` block in the post. Thank you!

TroySK commented:
I tried flashing it to an esp32dev but it seems to be rebooting. Here is log.
https://pastebin.com/AYeUAJts

ennetech commented:
> I tried flashing it to an esp32dev but it seems to be rebooting. Here is log. https://pastebin.com/AYeUAJts

What specific board you are using?

TroySK commented:
> > I tried flashing it to an esp32dev but it seems to be rebooting. Here is log. https://pastebin.com/AYeUAJts
> 
> What specific board you are using?

ESP32 DEVKITV1. It has the ESP-WROOM-32 chip. https://www.espboards.dev/esp32/esp32doit-devkit-v1/. Which one do you suggest? Did you test it on a C3 or S3 variant?

ennetech commented:
> > > I tried flashing it to an esp32dev but it seems to be rebooting. Here is log. https://pastebin.com/AYeUAJts
> > 
> > 
> > What specific board you are using?
> 
> ESP32 DEVKITV1. It has the ESP-WROOM-32 chip. https://www.espboards.dev/esp32/esp32doit-devkit-v1/. Which one do you suggest? Did you test it on a C3 or S3 variant?

I’m using the exact same board, I’ll try to compile again to see if something broke with last esphome versions

tach2004 commented:
Hi,
on this weekend i will also try to implement the dfrobot sen0623 to my esp32 S3 or esp32 firebeetle2 (wroom32E). hopefully it works.
I hope you continue working on the repo to integrate all sensors cleanly into ESPHome. I'll get started on it too. Maybe I can help a little.
And I hope it will be added to ESPHome at some point :)
Best regards.

tach2004 commented:
> I tried flashing it to an esp32dev but it seems to be rebooting. Here is log. https://pastebin.com/AYeUAJts

same here. doesnt matter if framework is idf or arduino.
i tried esphome v. 2024.9.1 and 2025.7.5.

any idea?

ennetech commented:
> > I tried flashing it to an esp32dev but it seems to be rebooting. Here is log. https://pastebin.com/AYeUAJts
> 
> same here. doesnt matter if framework is idf or arduino. i tried esphome v. 2024.9.1 and 2025.7.5.
> 
> any idea?

Maybe something with the wiring?

tach2004 commented:
Hhmm, I doubt it. As soon as I flash the code, the ESP no longer starts and goes into Safe Mode. 
If I comment out ‘dfrobot_sen0623,’ it starts again. Even if no cable is connected, it doesn’t start. 
That leads me to conclude that something is not correct with the code.

ennetech commented:
> Hhmm, I doubt it. As soon as I flash the code, the ESP no longer starts and goes into Safe Mode. If I comment out ‘dfrobot_sen0623,’ it starts again. Even if no cable is connected, it doesn’t start. That leads me to conclude that something is not correct with the code.

Looking at the logs:
Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:
[15:07:47]E (10785) task_wdt:  - loopTask (CPU 0)

This means that the loop is not receiving data and hangs indefinitely until watchdog resets the board

ennetech commented:
@tach2004 @TroySK i've tested and i experience the same issue when TX and RX are swapped, for reference in my esp32dev i am using TX2 and RX2 pins

TroySK commented:
> > Hhmm, I doubt it. As soon as I flash the code, the ESP no longer starts and goes into Safe Mode. If I comment out ‘dfrobot_sen0623,’ it starts again. Even if no cable is connected, it doesn’t start. That leads me to conclude that something is not correct with the code.
> 
> Looking at the logs: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time: [15:07:47]E (10785) task_wdt: - loopTask (CPU 0)
> 
> This means that the loop is not receiving data and hangs indefinitely until watchdog resets the board

Ahha! So I had stopped plugging in the sensor during flash and had thought it would boot. I connected the sensor and flashed again and it seems to work for most part of it. Breathing and Heart Rate sensors are still not working but that could be related to positioning of the device and me not being the its cone of detection. Will report it I get HR and BR to work as well.
<img width="1679" height="735" alt="image" src="https://github.com/user-attachments/assets/c5d049a5-f79c-4cae-8b3d-c1b01073688c" />

ennetech commented:
> > > Hhmm, I doubt it. As soon as I flash the code, the ESP no longer starts and goes into Safe Mode. If I comment out ‘dfrobot_sen0623,’ it starts again. Even if no cable is connected, it doesn’t start. That leads me to conclude that something is not correct with the code.
> > 
> > 
> > Looking at the logs: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time: [15:07:47]E (10785) task_wdt: - loopTask (CPU 0)
> > This means that the loop is not receiving data and hangs indefinitely until watchdog resets the board
> 
> Ahha! So I had stopped plugging in the sensor during flash and had thought it would boot. I connected the sensor and flashed again and it seems to work for most part of it. Breathing and Heart Rate sensors are still not working but that could be related to positioning of the device and me not being the its cone of detection. Will report it I get HR and BR to work as well. <img alt="image" width="1679" height="735" src="https://private-user-images.githubusercontent.com/2968254/505946211-c5d049a5-f79c-4cae-8b3d-c1b01073688c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE1NjAzNDksIm5iZiI6MTc2MTU2MDA0OSwicGF0aCI6Ii8yOTY4MjU0LzUwNTk0NjIxMS1jNWQwNDlhNS1mNzljLTRjYWUtOGIzZC1jMWIwMTA3MzY4OGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMjdUMTAxNDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWJiY2ExNDllNjIxMzM0ZWU4NGMyZjE4Nzk1NTIyNDg5NWY1MzYwMmEyMGVlY2FhZDQxNDk3MjUwMzUxMGE1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.OlM07CBllW37YyzcDk-eCRqtqrRPlJuxFmFAebYt6A0">

Remember to enable "Request rate", data should appear after 60s, device positioned in front of chest ~1mt

Also the device is operating in fall mode, that does not extract the data, let me see if i can push a button to switch mode for you

TroySK commented:
> > > Hhmm, I doubt it. As soon as I flash the code, the ESP no longer starts and goes into Safe Mode. If I comment out ‘dfrobot_sen0623,’ it starts again. Even if no cable is connected, it doesn’t start. That leads me to conclude that something is not correct with the code.
> > 
> > 
> > Looking at the logs: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time: [15:07:47]E (10785) task_wdt: - loopTask (CPU 0)
> > This means that the loop is not receiving data and hangs indefinitely until watchdog resets the board
> 
> Ahha! So I had stopped plugging in the sensor during flash and had thought it would boot. I connected the sensor and flashed again and it seems to work for most part of it. Breathing and Heart Rate sensors are still not working but that could be related to positioning of the device and me not being the its cone of detection. Will report it I get HR and BR to work as well. <img alt="image" width="1679" height="735" src="https://private-user-images.githubusercontent.com/2968254/505946211-c5d049a5-f79c-4cae-8b3d-c1b01073688c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE1NjA5NjEsIm5iZiI6MTc2MTU2MDY2MSwicGF0aCI6Ii8yOTY4MjU0LzUwNTk0NjIxMS1jNWQwNDlhNS1mNzljLTRjYWUtOGIzZC1jMWIwMTA3MzY4OGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMjdUMTAyNDIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjNjZmNkNjI2YTIwMGE5NDYzYThiZDVkNmM1NWE1OGFmMzRiMmE1N2FiZDZkZDI5NDQwZTZkZDdlMjExNjQ1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jK6xkQGFlPZ6jyowe0Q7sB617vLcVt8-4FDpZLFcb1E">

I had tried swapping the pins after reading some board have it wired incorrectly. I had tried different pins also previously and nothing worked and hence I had switched to flashing without the sensor attached. After the new ESPHome updates and your comment I tried again with similar configs and it worked the first time itself! Not sure why but thanks for pointing out the sensor data missing detail.

ennetech committed:
Add mode switch (sleep, fall)

ennetech commented:
> > > > Hhmm, I doubt it. As soon as I flash the code, the ESP no longer starts and goes into Safe Mode. If I comment out ‘dfrobot_sen0623,’ it starts again. Even if no cable is connected, it doesn’t start. That leads me to conclude that something is not correct with the code.
> > > 
> > > 
> > > Looking at the logs: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time: [15:07:47]E (10785) task_wdt: - loopTask (CPU 0)
> > > This means that the loop is not receiving data and hangs indefinitely until watchdog resets the board
> > 
> > 
> > Ahha! So I had stopped plugging in the sensor during flash and had thought it would boot. I connected the sensor and flashed again and it seems to work for most part of it. Breathing and Heart Rate sensors are still not working but that could be related to positioning of the device and me not being the its cone of detection. Will report it I get HR and BR to work as well. <img alt="image" width="1679" height="735" src="https://private-user-images.githubusercontent.com/2968254/505946211-c5d049a5-f79c-4cae-8b3d-c1b01073688c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE1NjA5NjEsIm5iZiI6MTc2MTU2MDY2MSwicGF0aCI6Ii8yOTY4MjU0LzUwNTk0NjIxMS1jNWQwNDlhNS1mNzljLTRjYWUtOGIzZC1jMWIwMTA3MzY4OGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMjdUMTAyNDIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjNjZmNkNjI2YTIwMGE5NDYzYThiZDVkNmM1NWE1OGFmMzRiMmE1N2FiZDZkZDI5NDQwZTZkZDdlMjExNjQ1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jK6xkQGFlPZ6jyowe0Q7sB617vLcVt8-4FDpZLFcb1E">
> 
> I had tried swapping the pins after reading some board have it wired incorrectly. I had tried different pins also previously and nothing worked and hence I had switched to flashing without the sensor attached. After the new ESPHome updates and your comment I tried again with similar configs and it worked the first time itself! Not sure why but thanks for pointing out the sensor data missing detail.

Added quick and dirty mode switch (you may need to press multiple times e/o restart the device after switch) also updated the configuration example in the first post

tach2004 commented:
Hello.
After a lot of trial and error and even a third esp32, I finally got it working :). There were several reasons that caused some confusion :( 
It wasn't actually the wiring. With rx/tx, I always test first to see if the two need to be reversed :)
It was partly due to the config in yaml because I had also hidden parts. I think that causes problems.
Currently, it still doesn't start cleanly every now and then when I completely cut the power supply. Then watchdog always comes on. But when I press the reset button on the esp, it starts.

The button you implemented "quick and dirty" is also working :) 

In any case, it's great that the sensor is now working. 
I hope you continue and integrate everything else the sensor can do into esphome :) That would be great. Unfortunately, I'm not the best at integrating things like that :)

TroySK commented:
@ennetech It's all working! Thank you 🤩
<img width="829" height="474" alt="image" src="https://github.com/user-attachments/assets/da9f1f51-aa06-4943-bebe-c1aa45ee1850" />

fredmorais commented:
I'm also using Devkit ESP32 and experiencing the boot problem, even if nothing is connected to the ESP, not even the C1001. Anyone got it fixed?

ennetech commented:
> I'm also using Devkit ESP32 and experiencing the boot problem, even if nothing is connected to the ESP, not even the C1001. Anyone got it fixed?

Connect the c1001 otherwise setup will fail and boot loop, triple check the wiring

fredmorais commented:
I tried that. Tried swapping the rx and tx pins as well and nothing works. Tried with the basic example of the library they provide for arduino and it was working.

ennetech commented:
> I tried that. Tried swapping the rx and tx pins as well and nothing works. Tried with the basic example of the library they provide for arduino and it was working.

Please provide:
- config
- wiring
- logs

fredmorais commented:
Here is the information:

It is connected to a Joy It ESP32 Devkit:
<img width="381" height="720" alt="image" src="https://github.com/user-attachments/assets/798f0b25-f0d4-43ea-b36d-6bf1c134be6b" />

The pins are connected like this:
ESP32    C1001
VIN         VIN
GND       GND
RX2        RX
TX2        TX

I also tried swapping RX and TX, the results are the same. The documentation for this particular ESP32 board is less than optimal, but using the Arduino library I can confirm that the RX2 and TX2 pins are 16 and 17. My ESPHome config:

```
ota:
  - platform: esphome
    password: <redacted>

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  use_address: <redacted>

mdns:
  disabled: true
captive_portal:

web_server:
  port: 80

# custom components
external_components:
  - source:
      type: git
      url: https://github.com/ennetech/esphome
      ref: component/dfrobot_sen0623
    components: [ dfrobot_sen0623 ]

# device config
uart:
  tx_pin: 17
  rx_pin: 16
  baud_rate: 115200
  debug:
    direction: BOTH
    dummy_receiver: false
    after:
      delimiter: "\n"
    sequence:
      - lambda: UARTDebug::log_string(direction, bytes);

# base component
dfrobot_sen0623:
  update_interval: 5s

sensor:
  - platform: dfrobot_sen0623
    heart_rate:
      name: "Heart rate"
    breath_rate:
      name: "Breath rate"
    human_distance:
      name: "Human distance"
    human_move_range:
      name: "Human move range"

text_sensor:
  - platform: dfrobot_sen0623
    status:
      name: "Operating mode"
    movement:
      name: "Movement type"
      
binary_sensor:
  - platform: dfrobot_sen0623
    presence:
      name: "Presence"

switch:
  - platform: dfrobot_sen0623
    type: request_rate
    name: "Request rate"
  - platform: dfrobot_sen0623
    type: hp_led
    name: "HP Led Enabled"
button:
  - platform: dfrobot_sen0623
    type: mode_fall
    name: "Fall mode"
  - platform: dfrobot_sen0623
    type: mode_sleep
    name: "Sleep mode"
  - platform: dfrobot_sen0623
    type: reset
    name: "Reset"
```

Logs:

```
[23:00:23]eE (10716) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:
[23:00:23]E (10716) task_wdt:  - loopTask (CPU 1)
[23:00:23]E (10716) task_wdt: Tasks currently running:
[23:00:23]E (10716) task_wdt: CPU 0: IDLE0
[23:00:23]E (10716) task_wdt: CPU 1: loopTask
[23:00:23]E (10716) task_wdt: Aborting.
[23:00:23]E (10716) task_wdt: Print CPU 1 backtrace
[23:00:23]
[23:00:23]
[23:00:23]
[23:00:23]
[23:00:23]Backtrace: 0x4000bfed:0x3ffbf330 0x40089b43:0x3ffbf340 0x4008912d:0x3ffbf360 0x400db226:0x3ffbf3a0 0x400d79dd:0x3ffbf3d0 0x400d7a7b:0x3ffbf420 0x400d7af9:0x3ffbf4c0 0x400e1ffa:0x3ffbf500 0x400e1c08:0x3ffbf520 0x400e154f:0x3ffbf540 0x400e4145:0x3ffbf580 0x400d7c67:0x3ffbf630
[23:00:23]
[23:00:23]
[23:00:23]
[23:00:23]
[23:00:23]ELF file SHA256: 6de783b1b
[23:00:23]
[23:00:23]Rebooting...
[23:00:23]ets Jun  8 2016 00:22:57
[23:00:23]
[23:00:23]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:00:23]configsip: 0, SPIWP:0xee
[23:00:23]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:00:23]mode:DIO, clock div:2
[23:00:23]load:0x3fff0030,len:6380
[23:00:23]ho 0 tail 12 room 4
[23:00:23]load:0x40078000,len:15916
[23:00:23]load:0x40080400,len:3860
[23:00:23]entry 0x40080630
[23:00:23]I (29) boot: ESP-IDF 5.5.1 2nd stage bootloader
[23:00:23]I (29) boot: compile time Nov 25 2025 22:59:02
[23:00:23]I (29) boot: Multicore bootloader
[23:00:23]I (30) boot: chip revision: v1.0
[23:00:23]I (33) boot.esp32: SPI Speed      : 40MHz
[23:00:23]I (37) boot.esp32: SPI Mode       : DIO
[23:00:23]I (40) boot.esp32: SPI Flash Size : 4MB
[23:00:23]I (44) boot: Enabling RNG early entropy source...
[23:00:23]I (48) boot: Partition Table:
[23:00:23]I (51) boot: ## Label            Usage          Type ST Offset   Length
[23:00:23]I (57) boot:  0 otadata          OTA data         01 00 00009000 00002000
[23:00:23]I (64) boot:  1 phy_init         RF data          01 01 0000b000 00001000
[23:00:23]I (70) boot:  2 app0             OTA app          00 10 00010000 001c0000
[23:00:23]I (77) boot:  3 app1             OTA app          00 11 001d0000 001c0000
[23:00:23]I (83) boot:  4 nvs              WiFi data        01 02 00390000 0006d000
[23:00:23]I (90) boot: End of partition table
[23:00:23]I (93) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=2369ch (145052) map
[23:00:23]I (150) esp_image: segment 1: paddr=000336c4 vaddr=3ffb0000 size=03de8h ( 15848) load
[23:00:23]I (156) esp_image: segment 2: paddr=000374b4 vaddr=40080000 size=08b64h ( 35684) load
[23:00:23]I (171) esp_image: segment 3: paddr=00040020 vaddr=400d0020 size=93d94h (605588) map
[23:00:23]I (378) esp_image: segment 4: paddr=000d3dbc vaddr=40088b64 size=0e264h ( 57956) load
[23:00:23]I (400) esp_image: segment 5: paddr=000e2028 vaddr=50000000 size=00020h (    32) load
[23:00:23]I (412) boot: Loaded app from partition at offset 0x10000
[23:00:23]I (412) boot: Disabling RNG early entropy source...
[23:00:23][I][logger:121]: Log initialized
[23:00:23][C][safe_mode:084]: Unsuccessful boot attempts: 3
[23:00:23][D][esp32.preferences:149]: Writing 1 items: 0 cached, 1 written, 0 failed
[23:00:23][I][app:073]: Running through setup()
[23:00:23][C][component:173]: Setup uart took 2ms
[23:00:23][C][component:173]: Setup preferences took 0ms
[23:00:23][C][component:173]: Setup uart took 0ms
[23:00:23][I][dfrobot_sen0623.component:320]: WAITING FOR INIT
[23:00:24][I][dfrobot_sen0623.component:314]: >> 01 83   0F   
[23:00:28]E (10717) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:
[23:00:28]E (10717) task_wdt:  - loopTask (CPU 1)
[23:00:28]E (10717) task_wdt: Tasks currently running:
[23:00:28]E (10717) task_wdt: CPU 0: IDLE0
[23:00:28]E (10717) task_wdt: CPU 1: loopTask
[23:00:28]E (10717) task_wdt: Aborting.
[23:00:28]E (10717) task_wdt: Print CPU 1 backtrace
[23:00:28]
[23:00:28]
[23:00:28]
[23:00:28]
[23:00:28]Backtrace: 0x4000bfed:0x3ffbf330 0x40089b43:0x3ffbf340 0x400894c9:0x3ffbf360 0x400db210:0x3ffbf3a0 0x400d79dd:0x3ffbf3d0 0x400d7a7b:0x3ffbf420 0x400d7af9:0x3ffbf4c0 0x400e1ffa:0x3ffbf500 0x400e1c08:0x3ffbf520 0x400e154f:0x3ffbf540 0x400e4145:0x3ffbf580 0x400d7c67:0x3ffbf630
[23:00:28]
[23:00:28]
[23:00:28]
[23:00:28]
[23:00:28]ELF file SHA256: 6de783b1b
[23:00:28]
[23:00:28]Rebooting...
[23:00:28]ets Jun  8 2016 00:22:57
[23:00:28]
[23:00:28]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:00:28]configsip: 0, SPIWP:0xee
[23:00:28]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:00:28]mode:DIO, clock div:2
[23:00:28]load:0x3fff0030,len:6380
[23:00:28]ho 0 tail 12 room 4
[23:00:28]load:0x40078000,len:15916
[23:00:28]load:0x40080400,len:3860
[23:00:28]entry 0x40080630
[23:00:28]I (29) boot: ESP-IDF 5.5.1 2nd stage bootloader
[23:00:28]I (29) boot: compile time Nov 25 2025 22:59:02
[23:00:29]I (29) boot: Multicore bootloader
[23:00:29]I (30) boot: chip revision: v1.0
[23:00:29]I (33) boot.esp32: SPI Speed      : 40MHz
[23:00:29]I (37) boot.esp32: SPI Mode       : DIO
[23:00:29]I (40) boot.esp32: SPI Flash Size : 4MB
[23:00:29]I (44) boot: Enabling RNG early entropy source...
[23:00:29]I (48) boot: Partition Table:
[23:00:29]I (51) boot: ## Label            Usage          Type ST Offset   Length
[23:00:29]I (57) boot:  0 otadata          OTA data         01 00 00009000 00002000
[23:00:29]I (64) boot:  1 phy_init         RF data          01 01 0000b000 00001000
[23:00:29]I (70) boot:  2 app0             OTA app          00 10 00010000 001c0000
[23:00:29]I (77) boot:  3 app1             OTA app          00 11 001d0000 001c0000
[23:00:29]I (83) boot:  4 nvs              WiFi data        01 02 00390000 0006d000
[23:00:29]I (90) boot: End of partition table
[23:00:29]I (93) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=2369ch (145052) map
[23:00:29]I (150) esp_image: segment 1: paddr=000336c4 vaddr=3ffb0000 size=03de8h ( 15848) load
[23:00:29]I (156) esp_image: segment 2: paddr=000374b4 vaddr=40080000 size=08b64h ( 35684) load
[23:00:29]I (171) esp_image: segment 3: paddr=00040020 vaddr=400d0020 size=93d94h (605588) map
[23:00:29]I (378) esp_image: segment 4: paddr=000d3dbc vaddr=40088b64 size=0e264h ( 57956) load
[23:00:29]I (400) esp_image: segment 5: paddr=000e2028 vaddr=50000000 size=00020h (    32) load
[23:00:29]I (412) boot: Loaded app from partition at offset 0x10000
[23:00:29]I (412) boot: Disabling RNG early entropy source...
[23:00:29][I][logger:121]: Log initialized
[23:00:29][C][safe_mode:084]: Unsuccessful boot attempts: 4
[23:00:29][D][esp32.preferences:149]: Writing 1 items: 0 cached, 1 written, 0 failed
[23:00:29][I][app:073]: Running through setup()
[23:00:29][C][component:173]: Setup uart took 1ms
[23:00:29][C][component:173]: Setup preferences took 0ms
[23:00:29][C][component:173]: Setup uart took 0ms
[23:00:29][I][dfrobot_sen0623.component:320]: WAITING FOR INIT
[23:00:30][I][dfrobot_sen0623.component:314]: >> 01 83   0F   
[23:00:34]E (10716) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:
[23:00:34]E (10716) task_wdt:  - loopTask (CPU 1)
[23:00:34]E (10716) task_wdt: Tasks currently running:
[23:00:34]E (10716) task_wdt: CPU 0: IDLE0
[23:00:34]E (10716) task_wdt: CPU 1: loopTask
[23:00:34]E (10716) task_wdt: Aborting.
[23:00:34]E (10716) task_wdt: Print CPU 1 backtrace
[23:00:34]
[23:00:34]
[23:00:34]
[23:00:34]
[23:00:34]Backtrace: 0x4000bfed:0x3ffbf330 0x40089b43:0x3ffbf340 0x4008912d:0x3ffbf360 0x400db226:0x3ffbf3a0 0x400d79dd:0x3ffbf3d0 0x400d7a7b:0x3ffbf420 0x400d7af9:0x3ffbf4c0 0x400e1ffa:0x3ffbf500 0x400e1c08:0x3ffbf520 0x400e154f:0x3ffbf540 0x400e4145:0x3ffbf580 0x400d7c67:0x3ffbf630
[23:00:34]
[23:00:34]
[23:00:34]
[23:00:34]
[23:00:34]ELF file SHA256: 6de783b1b
[23:00:34]
[23:00:34]Rebooting...
[23:00:34]ets Jun  8 2016 00:22:57
[23:00:34]
[23:00:34]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:00:34]configsip: 0, SPIWP:0xee
[23:00:34]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:00:34]mode:DIO, clock div:2
[23:00:34]load:0x3fff0030,len:6380
[23:00:34]ho 0 tail 12 room 4
[23:00:34]load:0x40078000,len:15916
[23:00:34]load:0x40080400,len:3860
[23:00:34]entry 0x40080630
[23:00:34]I (29) boot: ESP-IDF 5.5.1 2nd stage bootloader
[23:00:34]I (29) boot: compile time Nov 25 2025 22:59:02
[23:00:34]I (29) boot: Multicore bootloader
[23:00:34]I (30) boot: chip revision: v1.0
[23:00:34]I (33) boot.esp32: SPI Speed      : 40MHz
[23:00:34]I (37) boot.esp32: SPI Mode       : DIO
[23:00:34]I (40) boot.esp32: SPI Flash Size : 4MB
[23:00:34]I (44) boot: Enabling RNG early entropy source...
[23:00:34]I (48) boot: Partition Table:
[23:00:34]I (51) boot: ## Label            Usage          Type ST Offset   Length
[23:00:34]I (57) boot:  0 otadata          OTA data         01 00 00009000 00002000
[23:00:34]I (64) boot:  1 phy_init         RF data          01 01 0000b000 00001000
[23:00:34]I (70) boot:  2 app0             OTA app          00 10 00010000 001c0000
[23:00:34]I (77) boot:  3 app1             OTA app          00 11 001d0000 001c0000
[23:00:34]I (83) boot:  4 nvs              WiFi data        01 02 00390000 0006d000
[23:00:34]I (90) boot: End of partition table
[23:00:34]I (93) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=2369ch (145052) map
[23:00:34]I (150) esp_image: segment 1: paddr=000336c4 vaddr=3ffb0000 size=03de8h ( 15848) load
[23:00:34]I (156) esp_image: segment 2: paddr=000374b4 vaddr=40080000 size=08b64h ( 35684) load
[23:00:34]I (171) esp_image: segment 3: paddr=00040020 vaddr=400d0020 size=93d94h (605588) map
[23:00:34]I (378) esp_image: segment 4: paddr=000d3dbc vaddr=40088b64 size=0e264h ( 57956) load
[23:00:35]I (400) esp_image: segment 5: paddr=000e2028 vaddr=50000000 size=00020h (    32) load
[23:00:35]I (412) boot: Loaded app from partition at offset 0x10000
[23:00:35]I (412) boot: Disabling RNG early entropy source...
[23:00:35][I][logger:121]: Log initialized
[23:00:35][C][safe_mode:084]: Unsuccessful boot attempts: 5
[23:00:35][D][esp32.preferences:149]: Writing 1 items: 0 cached, 1 written, 0 failed
[23:00:35][I][app:073]: Running through setup()
[23:00:35][C][component:173]: Setup uart took 1ms
[23:00:35][C][component:173]: Setup preferences took 0ms
[23:00:35][C][component:173]: Setup uart took 0ms
[23:00:35][I][dfrobot_sen0623.component:320]: WAITING FOR INIT
[23:00:36][I][dfrobot_sen0623.component:314]: >> 01 83   0F   
```

Extra note: Also, I have no idea why but the IO1 and IO2 pins are not behaving as they should. I tried disabling the custom component in ESPHome and just use the 2 pins as binary inputs and they were either always on or off, depending on if I used invert or PULLUP or PULLDOWN pins (tested all the configurations, nothing worked), which is weird as this is fairly simple. I don't believe I have a faulty sensor as the LEDs light up when presence is detected and using the Arduino library also yielded good results.

Thank you for the support.

ennetech commented:
> [23:00:35][I][dfrobot_sen0623.component:320]: WAITING FOR INIT
> [23:00:36][I][dfrobot_sen0623.component:314]: >> 01 83   0F

definitely something with serial:
[23:00:35][I][dfrobot_sen0623.component:320]: WAITING FOR INIT
[23:00:36][I][dfrobot_sen0623.component:314]: >> 01 83   0F  

device is not receiving data, try to remove the UART debug and invert RX - TX again

fredmorais commented:
That yields very different results but still errors (found networks logs redacted):

```
[17:03:23]
[17:03:23]����|�#��AU�p!�,������������������������3f400020 size=2352ch (144684) map
[17:03:23]I (150) esp_image: segment 1: paddr=00033554 vaddr=3ffb0000 size=03de8h ( 15848) load
[17:03:23]I (156) esp_image: segment 2: paddr=00037344 vaddr=40080000 size=08cd4h ( 36052) load
[17:03:23]I (171) esp_image: segment 3: paddr=00040020 vaddr=400d0020 size=93610h (603664) map
[17:03:24]I (377) esp_image: segment 4: paddr=000d3638 vaddr=40088cd4 size=0e0f4h ( 57588) load
[17:03:24]I (400) esp_image: segment 5: paddr=000e1734 vaddr=50000000 size=00020h (    32) load
[17:03:24]I (411) boot: Loaded app from partition at offset 0x10000
[17:03:24]I (411) boot: Disabling RNG early entropy source...
[17:03:24][I][logger:121]: Log initialized
[17:03:24][C][safe_mode:084]: Unsuccessful boot attempts: 9
[17:03:24][D][esp32.preferences:149]: Writing 1 items: 0 cached, 1 written, 0 failed
[17:03:24][I][app:073]: Running through setup()
[17:03:24][C][component:173]: Setup uart took 1ms
[17:03:24][C][component:173]: Setup preferences took 0ms
[17:03:24][I][dfrobot_sen0623.component:320]: WAITING FOR INIT
[17:03:25][I][dfrobot_sen0623.component:314]: >> 01 83   0F   
[17:03:25][I][dfrobot_sen0623.component:314]: << FE 00   FA 00 FE 00 FF 00 08 C0 00 FD 00 FA 00 FC 00 FE 00 F4 00 00 FE 00 00 FC 00 F8 00 F6 00 00 FE FD 00 F0 00 FC 00 FD 00 FE 00 FE 00 FD 00 FF 00 FE 00 00 C0 05 FF 00 FF 00 FF 00 FE 00 FF 02 FF 00 03 00 FE 02 00 FF 03 FF 00 FE 08 FC 00 FC 01 FE 00 FC 00 FE 00 FC 00 FC 00 F4 00   
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:25][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:26][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: <<  
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:27][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:28][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: <<  
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29][I][dfrobot_sen0623.component:314]: << 
[17:03:29]E (10713) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:
[17:03:29]E (10713) task_wdt:  - loopTask (CPU 1)
[17:03:29]E (10713) task_wdt: Tasks currently running:
[17:03:29]E (10713) task_wdt: CPU 0: IDLE0
[17:03:29]E (10713) task_wdt: CPU 1: loopTask
[17:03:29]E (10713) task_wdt: Aborting.
[17:03:29]E (10713) task_wdt: Print CPU 1 backtrace
[17:03:29]
[17:03:29]
[17:03:29]
[17:03:29]
[17:03:29]Backtrace: 0x4000bfed:0x3ffbf320 0x40089b43:0x3ffbf330 0x4008912d:0x3ffbf350 0x400db217:0x3ffbf390 0x400d7995:0x3ffbf3c0 0x400d7a33:0x3ffbf410 0x400d7ab1:0x3ffbf4b0 0x400e1af2:0x3ffbf4f0 0x400e1700:0x3ffbf510 0x400e1047:0x3ffbf530 0x400e3b09:0x3ffbf570 0x400d7c1f:0x3ffbf620
[17:03:29]
[17:03:29]
[17:03:29]
[17:03:29]
[17:03:29]ELF file SHA256: 25816deea
[17:03:29]
[17:03:29]Rebooting...
[17:03:29]ets Jun  8 2016 00:22:57
[17:03:29]
[17:03:29]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[17:03:29]configsip: 0, SPIWP:0xee
[17:03:29]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[17:03:29]mode:DIO, clock div:2
[17:03:29]load:0x3fff0030,len:6380
[17:03:29]ho 0 tail 12 room 4
[17:03:29]load:0x40078000,len:15916
[17:03:29]load:0x40080400,len:3860
[17:03:29]entry 0x40080630
[17:03:29]I (29) boot: ESP-IDF 5.5.1 2nd stage bootloader
[17:03:29]I (29) boot: compile time Nov 29 2025 17:00:54
[17:03:29]I (29) boot: Multicore bootloader
[17:03:29]I (30) boot: chip revision: v1.0
[17:03:29]I (33) boot.esp32: SPI Speed      : 40MHz
[17:03:29]I (37) boot.esp32: SPI Mode       : DIO
[17:03:29]I (40) boot.esp32: SPI Flash Size : 4MB
[17:03:29]I (44) boot: Enabling RNG early entropy source...
[17:03:29]I (48) boot: Partition Table:
[17:03:29]I (51) boot: ## Label            Usage          Type ST Offset   Length
[17:03:29]I (57) boot:  0 otadata          OTA data         01 00 00009000 00002000
[17:03:29]I (64) boot:  1 phy_init         RF data          01 01 0000b000 00001000
[17:03:29]I (70) boot:  2 app0             OTA app          00 10 00010000 001c0000
[17:03:29]I (77) boot:  3 app1             OTA app          00 11 001d0000 001c0000
[17:03:29]I (83) boot:  4 nvs              WiFi data        01 02 00390000 0006d000
[17:03:29]I (90) boot: End of partition table
[17:03:29]I (93) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=2352ch (144684) map
[17:03:29]I (150) esp_image: segment 1: paddr=00033554 vaddr=3ffb0000 size=03de8h ( 15848) load
[17:03:29]I (156) esp_image: segment 2: paddr=00037344 vaddr=40080000 size=08cd4h ( 36052) load
[17:03:29]I (171) esp_image: segment 3: paddr=00040020 vaddr=400d0020 size=93610h (603664) map
[17:03:29]I (377) esp_image: segment 4: paddr=000d3638 vaddr=40088cd4 size=0e0f4h ( 57588) load
[17:03:29]I (400) esp_image: segment 5: paddr=000e1734 vaddr=50000000 size=00020h (    32) load
[17:03:29]I (411) boot: Loaded app from partition at offset 0x10000
[17:03:29]I (411) boot: Disabling RNG early entropy source...
[17:03:29][I][logger:121]: Log initialized
[17:03:30][C][safe_mode:084]: Unsuccessful boot attempts: 10
[17:03:30][D][esp32.preferences:149]: Writing 1 items: 0 cached, 1 written, 0 failed
[17:03:30][E][safe_mode:096]: Boot loop detected
[17:03:30][E][component:315]: safe_mode set Error flag: unspecified
[17:03:30][I][app:073]: Running through setup()
[17:03:30][C][component:173]: Setup web_server_base took 0ms
[17:03:30][C][component:173]: Setup captive_portal took 1ms
[17:03:30][C][wifi:344]: Starting
[17:03:30][C][wifi:344]:  Local MAC: 7C:9E:BD:39:B4:4C
[17:03:30][I][wifi:305]: Starting scan
[17:03:30][D][wifi:888]: Starting scan
[17:03:30][C][component:173]: Setup wifi took 102ms
[17:03:30][C][component:173]: Setup esphome.ota took 2ms
[17:03:30][C][component:173]: Setup safe_mode took 0ms
[17:03:30][I][app:120]: setup() finished successfully!
[17:03:30][W][safe_mode:109]: SAFE MODE IS ACTIVE
[17:03:30][W][component:307]: wifi set Warning flag: scanning for networks
[17:03:30][I][app:190]: ESPHome version 2025.11.2 compiled on Nov 29 2025, 16:59:35
[17:03:30][C][logger:261]: Logger:
[17:03:30][C][logger:261]:  Max Level: DEBUG
[17:03:30][C][logger:261]:  Initial Level: DEBUG
[17:03:30][C][logger:267]:   Log Baud Rate: 115200
[17:03:30][C][logger:267]:  Hardware UART: UART0
[17:03:30][C][logger:274]:   Task Log Buffer Size: 768
[17:03:30][C][captive_portal:122]: Captive Portal:
[17:03:30][C][wifi:1062]: WiFi:
[17:03:30][C][wifi:1062]:  Connected: NO
[17:03:30][C][wifi:827]:   Local MAC: [redacted]
[17:03:30][C][wifi:838]:   SSID: [redacted]
[17:03:30][C][wifi:838]:  BSSID: [redacted]
[17:03:30][C][wifi:838]:  Hostname: 'esp32-c1001'
[17:03:30][C][wifi:838]:  Signal strength: -127 dB ××××
[17:03:30][C][wifi:838]:  Channel: 1
[17:03:30][C][wifi:838]:  Subnet: 0.0.0.0
[17:03:30][C][wifi:838]:  Gateway: 0.0.0.0
[17:03:30][C][wifi:838]:  DNS1: 0.0.0.0
[17:03:30][C][wifi:838]:  DNS2: 0.0.0.0
[17:03:30][C][esphome.ota:093]: Over-The-Air updates:
[17:03:30][C][esphome.ota:093]:  Address: [redacted]
[17:03:30][C][esphome.ota:093]:  Version: 2
[17:03:30][C][esphome.ota:100]:   Password configured
[17:03:30][C][safe_mode:018]: Safe Mode:
[17:03:30][C][safe_mode:018]:  Successful after: 60s
[17:03:30][C][safe_mode:018]:  Invoke after: 10 attempts
[17:03:30][C][safe_mode:018]:  Duration: 300s
[17:03:30][W][safe_mode:032]: SAFE MODE IS ACTIVE
[17:03:34][W][component:454]: wifi took a long time for an operation (326 ms)
[17:03:34][W][component:457]: Components should block for at most 30 ms
[17:03:37][I][wifi:1079]: Connected
[17:03:37][C][wifi:827]:   Local MAC: [redacted]
[17:03:37][C][wifi:834]:   IP Address: [redacted]
[17:03:37][C][wifi:838]:   SSID: [redacted]
[17:03:37][C][wifi:838]:  BSSID: [redacted]
[17:03:37][C][wifi:838]:  Hostname: 'esp32-c1001'
[17:03:37][C][wifi:838]:  Signal strength: -68 dB ▂▄▆█
[17:03:37][C][wifi:838]:  Channel: 1
[17:03:37][C][wifi:838]:  Subnet: [redacted]
[17:03:37][C][wifi:838]:  Gateway: [redacted]
[17:03:37][C][wifi:838]:  DNS1: [redacted]
[17:03:37][C][wifi:838]:  DNS2: [redacted]
[17:03:37][W][component:337]: wifi cleared Warning flag
[17:04:30][I][safe_mode:042]: Boot seems successful; resetting boot loop counter
[17:04:30][D][esp32.preferences:149]: Writing 1 items: 1 cached, 0 written, 0 failed
```

fredmorais commented:
I think I have reached a dead end. I got it to work momentarily with a certain wiring. However, it was very unstable. Could turn on/off the LED but presence wasn't being detected. Tried to go back to plain Arduino and their default library and couldn't make it work. Then moving the headers was helping, which made me think they could be poorly soldered. Tried resoldering and now nothing works. Maybe it's a faulty sensor, maybe I fried it with experimentations, have no idea.

Anyways, will look into an alternative but this is definity not going to take any more of my time. Appreciate all the help!

cianpdx commented:
doesn't seem to with work esp8266, but works with ep32 pretty well. 
getting lots of blocking warning/long execution times which seems odd with fast serial and low amount of data. 

only other issue i see is when the subject leaves the sensing area, the stats are held at the last measurement instead of dropping to zero or N/A. 

is this just a trait of the sensor?

ennetech commented:
> doesn't seem to with work esp8266, but works with ep32 pretty well. getting lots of blocking warning/long execution times which seems odd with fast serial and low amount of data.
> 
> only other issue i see is when the subject leaves the sensing area, the stats are held at the last measurement instead of dropping to zero or N/A.
> 
> is this just a trait of the sensor?

Regarding the device, I have tested only with esp32
I'll check next week and let you know for the data persistence
